### PR TITLE
Ignore environment directories at workspace root

### LINKED ISSUE #172: The extension should ignore environment directories that are opened directly.
If your workspace root is a VectorCAST environment directory, the test pane will have an environment node with no name, and some commands like Re-Build Environment will fail because the directory cannot be deleted when it is open in the Workspace!

This edge case should not have been supported in the first place.

### PR DESCRIPTION
**Pull request by johnpaliotta**
Also fixed an "extra" update that was being done on insert ATG tests

Closes #172

### CODE REVIEW

**Comment by aytey** [comment]
Can we add a notification that says something like "the current workspace is a VectorCAST environment; to open-up this environment, please open the parent folder as the workspace"?

File: src/testPane.ts [hunk 1]
```diff
@@ -292,10 +305,19 @@ function getEnvironmentList(baseDirectory: string): string[] {
...
   // so turn this into a list of the enclosing directories only
   let returnList: string[] = [];
+  const workspaceFolderList = getWorkspaceFolderList();
   for (const file of fileList) {
     // note that glob always uses / as path separator ...
     if (!file.includes(".BAK")) {
-      returnList.push(path.dirname(file));
+      const candidatePath = path.dirname(file);
+      // we don't want to support users who add an enviro
+      // directory to the workspace - someone did this :()
```
Comment (aytey) [inline src/testPane.ts:314]: Do you want a `:)` or a `:(` here? :D

File: src/testPane.ts [hunk 2]
```diff
@@ -292,10 +305,19 @@ function getEnvironmentList(baseDirectory: string): string[] {
...
+      if (!workspaceFolderList?.includes(candidatePath)) {
+        returnList.push(candidatePath);
+      } else {
+        vectorMessage (`Ignoring environment: ${candidatePath} ...`);
+        vectorMessage (`   environments should not be added to the workspace.`);
```
Comment (aytey) [inline src/testPane.ts:319]: "environments should not be opened as the workspace root"?

The current wording reads like "never open up a workspace that contains environments" (which is clearly wrong).

**Comment by johnpaliotta** [comment]
> @johnpaliotta do you want a test for this?

not needed

**Review by aytey — APPROVED** [review]
